airflow/providers/hashicorp/_internal_client/vault_client.py

```diff
@@ -191,8 +191,21 @@ def __init__(
         self.radius_secret = radius_secret
         self.radius_port = radius_port
 
+    @property
+    def client(self):
+        """
+        Wrapper function to invalidate cache
+
+        :rtype: hvac.Client
+        :return: Vault Client
+
+        """
+        if '_client' in self.__dict__ and not self._client.is_authenticated():
+            del self.__dict__['_client']
+        return self._client
```

Comment:
```suggestion
        client = self._client
        # Invalidate so we can provide auth to get an authenticated client later.
        if not client.is_authenticated():
            del self.__dict__['_client']        
        return client
```

I wonder if this is easier to undersstand.

Also, modifying `__dict__` to invalidate the cached property feels a bit hacky to me (although `cached_property` is well-documented to use `__dict__` so it’s probably OK).

Maybe it’s better to do something like this?

```python
@functools.lru_cache()
def _create_client(self) -> hvac.Client:
    ...

@property
def client(self) -> hvac.Client:
    client = self._create_client()
    if not client.is_authenticated():
        self._create_client.cache_clear()
    return client
```

Reply:
I'm not sure it is easier to understand. The assignment here is implicitly making sure that `_client` is inside `__dict__`, whereas the `if '_client' in self.__dict__` is explicit. I agree that modifying `__dict__` is hacky, but as mentioned this is just how `cached_property` works and is documented. 

I'm not sure that we want to use the `lru_cache`, as it could be semantically confusing here. There is no benefit to using a lru except for the convenient `.cache_clear()`. I think adding a comment is definitely a good idea though.

Reply:
Could you add a comment then?

Reply:
You bet!